### PR TITLE
Implemented SkiaPImage class and image APIs for skia

### PR DESCRIPTION
**Pull request by tushar5526**
Related to #379 
- SkiaPImage is completed except from blend
- Implemented image APIs for Skia Renderer

### CODE REVIEW

**Comment by tushar5526** [comment]
Hey @ziyaointl 

Please review this PR. I have implemented all the features that were there in p5py. There are other functionalities that I think is possible as well, I will be covering them later or make an issue for the community to involve as well.

File: p5/sketch/Skia2DRenderer/renderer2d.py [hunk 1]
```diff
@@ -704,6 +707,42 @@ def end_shape(
...
+    def create_image(self, width, height):
+        return SkiaPImage(width, height)
+
+    def image(self, pimage, x, y, w=None, h=None):
+        if self.style.image_mode == constants.CORNER:
+            if w and h:
+                pimage.size = (w, h)
+        elif self.style.image_mode == constants.CORNERS:
+            pimage.size = (w - x, h - y)
+        elif self.style.image_mode == constants.CORNERS:
```
Comment (ziyaointl) [inline p5/sketch/Skia2DRenderer/renderer2d.py:720]: This condition and the one above look like they are the same.

File: p5/sketch/Skia2DRenderer/image.py [hunk 2]
```diff
@@ -0,0 +1,118 @@
...
+
+
+class SkiaPImage(PImage):
+    def __init__(self, width, height, pixels=None):
+        self._width = width
+        self._height = height
+        self.pixels = (
+            pixels
+            if pixels is not None
+            else np.zeros((width, height, 4), dtype=np.uint8)
```
Comment (ziyaointl) [inline p5/sketch/Skia2DRenderer/image.py:15]: I think it's useful to create a constant for 4 and name it something descriptive.

File: p5/sketch/Skia2DRenderer/image.py [hunk 3]
```diff
@@ -0,0 +1,118 @@
...
+        :param image: Image to be used as make
+        :type image: SkiaPImage
+        """
+        with skia.Surface(self.pixels) as canvas:
+            canvas.drawImage(image.surface.makeImageSnapshot())
+
+    def filter(self, kind, param=0.5):
+        if kind == constants.THRESHOLD:
+            threshold = param * 255
+            mask = np.dot(self.pixels[..., :3], [0.2989, 0.5870, 0.1140]) < threshold
```
Comment (ziyaointl) [inline p5/sketch/Skia2DRenderer/image.py:64]: Could you add a reference to these numbers?

**Review by ziyaointl — COMMENTED** [review]
Looking good! Thanks for adding documentation as well

File: p5/sketch/Skia2DRenderer/renderer2d.py (hunk 1 above)
Comment (tushar5526) [reply p5/sketch/Skia2DRenderer/renderer2d.py:720]: Oops, that should be `CENTER`. Thanks!

File: p5/sketch/Skia2DRenderer/image.py (hunk 2 above)
Comment (tushar5526) [reply p5/sketch/Skia2DRenderer/image.py:15]: What should I name it ? `RGBA_CHANNELS` ?
Do you have anything in mind?

File: p5/sketch/Skia2DRenderer/image.py (hunk 3 above)
Comment (tushar5526) [reply p5/sketch/Skia2DRenderer/image.py:64]: https://www.baeldung.com/cs/convert-rgb-to-grayscale#3-luminosity-method

This method is being used everywhere to convert RGB images to GREY

Comment (ziyaointl) [reply p5/sketch/Skia2DRenderer/image.py:65]: Ah I see. Thank you!

File: p5/sketch/Skia2DRenderer/image.py (hunk 2 above)
Comment (ziyaointl) [reply p5/sketch/Skia2DRenderer/image.py:15]: I think `RGBA_CHANNELS` works well!

**Review by ziyaointl — APPROVED** [review]

**Pull request merged**